packages/at_secondary_server/test/sync_unit_test.dart

```diff
@@ -0,0 +1,724 @@
+        await secondaryPersistenceStore!
+            .getSecondaryKeyStore()
+            ?.put('@alice:phone.wavi@alice', AtData()..data = '8897896766');
+        var syncProgressiveVerbHandler = SyncProgressiveVerbHandler(
+            secondaryPersistenceStore!.getSecondaryKeyStore()!);
+        var response = Response();
+        var inBoundSessionId = '_6665436c-29ff-481b-8dc6-129e89199718';
+        var atConnection = InboundConnectionImpl(null, inBoundSessionId);
+        atConnection.metaData.isAuthenticated = true;
+        var syncVerbParams = HashMap<String, String>();
+        syncVerbParams.putIfAbsent(AT_FROM_COMMIT_SEQUENCE, () => '-1');
+        await syncProgressiveVerbHandler.processVerb(
+            response, syncVerbParams, atConnection);
+        List syncResponse = jsonDecode(response.data!);
+
+        expect(syncResponse[0]['atKey'], 'public:country.wavi@alice');
+        expect(syncResponse[0]['commitId'], 1);
+        expect(syncResponse[0]['operation'], '+');
+        expect(syncResponse[0]['metadata']['version'], '0');
+
+        expect(syncResponse[1]['atKey'], '@bob:file-transfer.mosphere@alice');
+        expect(syncResponse[1]['commitId'], 2);
+        expect(syncResponse[1]['operation'], '+');
+        expect(syncResponse[1]['metadata']['version'], '0');
+
+        expect(syncResponse[2]['atKey'], 'firstname.buzz@alice');
+        expect(syncResponse[2]['commitId'], 3);
+        expect(syncResponse[2]['operation'], '+');
+        expect(syncResponse[2]['metadata']['version'], '0');
+
+        expect(syncResponse[3]['atKey'], '@alice:phone.wavi@alice');
+        expect(syncResponse[3]['commitId'], 4);
+        expect(syncResponse[3]['operation'], '*');
+      });
+
+      test(
+          'test to verify only entries matching the regex are added to sync response',
+          () async {
+        /// Preconditions:
+        /// The Commit Log Keystore contains the keys
+        ///
+        /// Assertions
+        /// The sync response contains the keys that matches the regex in sync request
+        var syncProgressiveVerbHandler = SyncProgressiveVerbHandler(
+            secondaryPersistenceStore!.getSecondaryKeyStore()!);
+        var response = Response();
+        var inBoundSessionId = '_6665436c-29ff-481b-8dc6-129e89199718';
+        var atConnection = InboundConnectionImpl(null, inBoundSessionId);
+        atConnection.metaData.isAuthenticated = true;
+        var syncVerbParams = HashMap<String, String>();
+        syncVerbParams.putIfAbsent(AT_FROM_COMMIT_SEQUENCE, () => '-1');
+        syncVerbParams.putIfAbsent('regex', () => 'buzz');
+        await syncProgressiveVerbHandler.processVerb(
+            response, syncVerbParams, atConnection);
+
+        List syncResponse = jsonDecode(response.data!);
+
+        expect(syncResponse.length, 2);
+        // As per design, all the public keys are not filtered when matching with regex.
+        expect(syncResponse[0]['atKey'], 'public:country.wavi@alice');
```

Comment:
How is this key matching the regex 'buzz'?

Reply:
When matching the regex, we look for keys that match the regex or if the key falls under the special key.

Below is the snippet
```dart
bool _acceptKey(String atKey, String regex) {
    return _isRegexMatches(atKey, regex) || _isSpecialKey(atKey);
  }
```

In this case, the key does not match the regex but matches the special key - because the key is public.

Below is the snippet for _specialKey method:
```dart
bool _isSpecialKey(String atKey) {
    return atKey.contains(AT_ENCRYPTION_SHARED_KEY) ||
        atKey.startsWith('public:') ||
        atKey.contains(AT_PKAM_SIGNATURE) ||
        atKey.contains(AT_SIGNING_PRIVATE_KEY);
  }
```

Reply:
That logic seems incorrect. But we will deal with that later.